### PR TITLE
fix(server): Allow passwordless users when oauth enabled

### LINKED ISSUE #13274: Cannot create passwordless user, even though passwords are disabled
### The bug

Unable to create accounts without providing a password, even though I have password logins disabled and I'm using OAuth2 (working)

### The OS that Immich Server is running on

Debian

### Version of Immich Server

v1.117.0

### Version of Immich Mobile App

N/A

### Platform with the issue

- [ ] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
N/A
```

### Your .env content

```Shell
N/A
```

## Steps

1. Go to `Administration`
2. `Settings`
3. `Auth settings`
4. `Oauth` -> enabled and working
5. `Password Login` -> disabled
6. Go to `Users`
7. `Create user`
8. Provide `Email`, disable `Require user to change password on first login`
9. Click `Create`

## Expected

User gets created and can auth with OAuth2, assuming user has account with matching email with the OAuth2 provider.

## Actual

Error `Please fill in this field` appears at the password field.

### Relevant log output

```shell
N/A
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jedi04**
This PR fixes : #13274 

It allows creation of passwordless users when Oauth is enabled in config.

### CODE REVIEW

**Review by jrasm91 — COMMENTED** [review]
Looks good! You can use the $featureFlag.oauthEnabled store instead I think

**Comment by jedi04** [comment]
> Looks good! You can use the $featureFlag.oauthEnabled store instead I think

@jrasm91 Thanks for the suggestion. I have updated it to use featureFlags.

**Review by danieldietzler — APPROVED** [review]

File: server/src/dtos/user.dto.ts [hunk 1]
```diff
@@ -62,7 +62,6 @@ export class UserAdminCreateDto {
   @Transform(toEmail)
   email!: string;
 
-  @IsNotEmpty()
```
Comment (jrasm91) [inline server/src/dtos/user.dto.ts:65]: I think this just allows empty strings, but it should probably have Optional({ nullable: true }) and still keep the is not empty. We should also test that you cannot login with an empty string of password login is enabled still

Comment (jedi04) [reply server/src/dtos/user.dto.ts:65]: We have set password to be string in the user entity. So would it be okay to insert null values?
https://github.com/immich-app/immich/blob/3f663106e8093824617128d7558423284c4741cd/server/src/entities/user.entity.ts#L33

And during login we have added check for not allowing empty password
https://github.com/immich-app/immich/blob/3f663106e8093824617128d7558423284c4741cd/server/src/services/auth.service.ts#L329

Comment (jrasm91) [reply server/src/dtos/user.dto.ts:65]: Oh, I see that we don't allow null in the database. We can change that later I think. We'll allow empty now, and move to `null` in the future.
